File: CSharp/Tests/ValidParentheses.cs
using Solutions;

namespace Tests;

public class ValidParenthesesTest {
    [Fact]
    public void TestValidParentheses_Case1() {
        string s = "[]";
        bool want = true;

        bool result = ValidParentheses.IsValid(s);
        Assert.Equal(want, result);
    }

    [Fact]
    public void TestValidParentheses_Case2() {
        string s = "()[]{}";
        bool want = true;

        bool result = ValidParentheses.IsValid(s);
        Assert.Equal(want, result);
    }

    [Fact]
    public void TestValidParentheses_Case3() {
        string s = "(]";
        bool want = false;

        bool result = ValidParentheses.IsValid(s);
        Assert.Equal(want, result);
    }
}

File: Rust/tests/valid_parentheses.rs
#[cfg(test)]
mod tests {
    use leetcode::valid_parentheses::*;

    #[test]
    fn test_valid_parentheses_case1() {
        let s = "()".to_string();
        let want = true;

        let result = is_valid(s);

        assert_eq!(result, want);
    }

    #[test]
    fn test_valid_parentheses_case2() {
        let s = "()[]{}".to_string();
        let want = true;

        let result = is_valid(s);

        assert_eq!(result, want);
    }

    #[test]
    fn test_valid_parentheses_case3() {
        let s = "(]".to_string();
        let want = false;

        let result = is_valid(s);

        assert_eq!(result, want);
    }

    #[test]
    fn test_valid_parentheses_case4() {
        let s = "){".to_string();
        let want = false;

        let result = is_valid(s);

        assert_eq!(result, want);
    }

    #[test]
    fn test_valid_parentheses_case5() {
        let s = "][]".to_string();
        let want = false;

        let result = is_valid(s);

        assert_eq!(result, want);
    }
}
